ynr/apps/elections/templates/elections/ballot_view.html

```diff
@@ -90,46 +93,61 @@ <h1>Candidates for {{ ballot.post.label }}</h1>
                 {% if candidate.elected %}</strong>{% endif %}
               </a>
             </td>
-            <td>{{ candidate.party.name }}</td>
+            <td>
+              {{ candidate.party.name }}
+            </td> 
+            <td>
+              {% for membership in candidate.person.memberships.all %}
```

Comment:
We dont want to display all `previous_party_affiliations` for any previous candidacy (membership) - only the previous party affiliations related to the candidacy for this ballot. So the multiple forloops can be removed to something more like the below:
```
{% if ballot.is_welsh_run %}
    <td>
        <ul class="previous-party>"
        {% for party in candidate.previous_party_affiliations.all %}
            <li>{{ party.name }}</li>
        {% endfor %}
        </ul>
    </td>  
{% endif %}   
```
I've also changed the HTML a bit there which I think should keep the number of DB queries down, and also only render elements when necessary so we dont end up with empty ones for non-welsh ballots